### PR TITLE
chore: setup vitest to replace jest

### PR DESCRIPTION
**Pull request by BugGambit**
One step of breaking up this PR into smaller pieces: https://github.com/cognitedata/cognite-sdk-js/pull/1128

Jira: https://cognitedata.atlassian.net/browse/AUTH-3067

### CODE REVIEW

**Review by samir-ajdarpasic — APPROVED** [review]

File: packages/core/src/__tests__/multiError.unit.spec.ts [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
 
 describe('Cognite multi error', () => {
   test('create with 2 fails and 1 success', () => {
     const errMsg = 'createAssets.arg0.items: size must be between 1 and 1000';
     const nestedErr = new CogniteError(errMsg, 400, 'r1', {
-      missing: ['something'],
+      missing: [
+        {
+          id: 'something',
+        },
```
Comment (BugGambit) [inline packages/core/src/__tests__/multiError.unit.spec.ts:12]: This is the correct format according to the API reference

File: packages/stable/src/__tests__/api/containers.int.spec.ts [hunk 2]
```diff
@@ -120,34 +118,46 @@ describe('Containers integration test', () => {
...
-      {
-        space: TEST_SPACE_NAME,
-        externalId: containerCreationDefinition2.externalId,
-      },
-    ]);
-    expect(response.items).toHaveLength(2);
+  it(
+    'should successfully delete Containers',
+    async () => {
+      const response = await client.containers.delete([
```
Comment (BugGambit) [inline packages/stable/src/__tests__/api/containers.int.spec.ts:124]: Just indent as we set timeout as the third argument to `it`

File: packages/stable/src/__tests__/api/datamodels.int.spec.ts [hunk 3]
```diff
@@ -158,24 +158,34 @@ describe('Data models integration test', () => {
...
-    expect(
-      dataModels.items.find(
-        (dm) => dm.externalId === datamodelCreationDefinition.externalId
-      )
-    ).toBeUndefined();
-  });
+  it(
+    'should successfully delete datamodels',
+    async () => {
+      const response = await client.dataModels.delete([
```
Comment (BugGambit) [inline packages/stable/src/__tests__/api/datamodels.int.spec.ts:164]: Just indent as we send a third argument to `it`

**Review by NilsBarlaug — APPROVED** [review]
I am not able to actually understand the details and nuances here, but at a high level it LGTM 🚀

**Pull request merged**